internal/graph/check.go

```diff
@@ -1156,7 +1154,15 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 			)
 			defer filteredIter.Stop()
 
-			return c.checkUsersetSlowPath(ctx, req, filteredIter)
+			resolver := c.checkUsersetSlowPath
+
+			if c.optimizationsEnabled {
+				if typesys.UsersetCanFastPath(directlyRelatedUsersetTypes, tuple.GetType(reqTupleKey.GetUser())) {
+					resolver = c.checkUsersetFastPath
+				}
+			}
```

Comment:
We can reduce an indentation level here if we use &&.
```suggestion
	if c.optimizationsEnabled &&
		typesys.UsersetCanFastPath(
			directlyRelatedUsersetTypes,
			tuple.GetType(reqTupleKey.GetUser()),
		) {
		resolver = c.checkUsersetFastPath
	}
```

Reply:
thanks for the suggestion, i think the current way is a bit more clear.